File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def helper(i, curr, cSum):
            if cSum == target:
                res.append(curr[:])
                return
            if i >= len(nums) or cSum > target:
                return
            
        # I want each number to include itself and all numbers ahead of it
            # This will include itself in the addition
            curr.append(nums[i])
            helper(i,curr,cSum+nums[i])

            curr.pop()
            helper(i+1,curr,cSum)
        
        helper(0,[],0)

        return res

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l=0, r=0;
        int min=0, currSum = 0;
        while (r < nums.length) {
            currSum += nums[r];
            r++;
            while (currSum >= target && l < r) {
                min = min==0 ? r-l : Math.min(min, r-l);
                currSum -= nums[l]; 
                l++;
            }
            // r++;
        }

        return min;
    }
}